core/src/main/java/io/aiven/kafka/tieredstorage/RemoteStorageManager.java

```diff
@@ -202,9 +202,9 @@ public void copyLogSegmentData(final RemoteLogSegmentMetadata remoteLogSegmentMe
             uploadIndexFile(remoteLogSegmentMetadata, leaderEpoch, LEADER_EPOCH);
         } catch (final StorageBackendException | IOException e) {
             throw new RemoteStorageException(e);
+        } finally {
+            metrics.recordSegmentCopyTime(startedMs, time.milliseconds());
         }
-
-        metrics.recordSegmentCopyTime(startedMs, time.milliseconds());
```

Comment:
We need to decide if we count failed attempts here and in other places. In the current code they aren't counted as the `record*` is called outside of `try`.

Reply:
Sure, we can take this on a separate PR